File: src/test/seu/Minisys32Assembler/ASMFileTest.java
package test.seu.Minisys32Assembler;

import com.seu.Minisys32Assembler.ASMFile;
import com.seu.Minisys32Assembler.Instruction;
import org.junit.Ignore;
import org.junit.Test;

import static org.junit.Assert.*;

public class ASMFileTest {

    @Test
    @Ignore
    public void readCode() throws Exception {
        ASMFile asmFile = new ASMFile("input/assemblerTest.asm");
        asmFile.instructionCodes.forEach(System.out::println);
    }

}

File: src/test/seu/MiniCCompiler/AsmFileTest.kt
package test.seu.MiniCCompiler

import com.seu.MiniCCompiler.Lexer
import com.seu.MiniCCompiler.Parser
import com.seu.MiniCCompiler.gen.AsmFile
import com.seu.MiniCCompiler.gen.CodeGenerator
import org.junit.BeforeClass
import org.junit.Ignore
import org.junit.Test

class AsmFileTest {

    companion object {
        private lateinit var asmFile: AsmFile

        @BeforeClass
        @JvmStatic
        fun constructor() {
            val lexer = Lexer("resource/compilerTest.c")
            val parser = Parser(lexer)
            parser.yyparse()
            asmFile = AsmFile(parser.generator)
        }
    }

    @Ignore
    @Test
    fun writeFile() {
        asmFile.writeFile("target/main.asm")
    }
}